akd/src/append_only_zks.rs

```diff
@@ -60,6 +60,106 @@ impl From<InsertMode> for NodeHashingMode {
     }
 }
 
+/// A set of nodes to be inserted into the tree. This abstraction denotes
+/// whether the nodes are binary searchable (i.e. all nodes have the same label
+/// length, and are sorted).
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) enum InsertionSet {
+    BinarySearchable(Vec<Node>),
+    Unsorted(Vec<Node>),
+}
+
+impl Deref for InsertionSet {
+    type Target = Vec<Node>;
+
+    fn deref(&self) -> &Self::Target {
+        match self {
+            InsertionSet::BinarySearchable(nodes) => nodes,
+            InsertionSet::Unsorted(nodes) => nodes,
+        }
+    }
+}
+
+impl From<Vec<Node>> for InsertionSet {
+    fn from(mut nodes: Vec<Node>) -> Self {
+        if !nodes.is_empty()
+            && nodes
+                .iter()
+                .all(|node| node.label.label_len == nodes[0].label.label_len)
+        {
+            nodes.sort();
```

Comment:
since we know we can't have duplicates, unstable sort is faster here, we should use it

Reply:
I think we would have been fine even with duplicates, thanks for the suggestion!